zerver/lib/actions.py

```diff
@@ -3153,6 +3153,18 @@ def do_change_is_admin(user_profile: UserProfile, value: bool,
                                  is_admin=value))
         send_event(event, active_user_ids(user_profile.realm_id))
 
+def do_change_is_guest(user_profile: UserProfile, value: bool) -> None:
+    if user_profile.is_realm_admin and value:
+        raise AssertionError("Admin can't be guest")
```

Comment:
Generally, we prefer to do this sort of check in the view function; partially, it's a naming convention (`do_foo` assumes that all checks have already been done; `check_foo` does not).  But also it's because this check is insufficient; we need to in the view function check for both (1) making a guest an admin and (2) making an admin a guest, and it's easier to do that there.

(And we should have tests for both of those two corner cases).

Reply:
Thanks! @timabbott.Got it, will soon update these changes and also write tests.